src/plugin.js

```diff
@@ -315,6 +325,7 @@ const contribAdsPlugin = function(options) {
   };
 
   player.ads._state = new BeforePreroll(player);
+  player.ads._state.init(player);
```

Comment:
Ah interesting, was this causing an issue or was it just something you noticed?

Reply:
Mostly something I noticed, I wanted to set terminate at the earliest point possible in BeforePreroll but felt putting it in the constructor wasn't right

Reply:
Yup makes total sense.